libsolidity/interface/OptimiserSettings.h

```diff
@@ -0,0 +1,72 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author Alex Beregszaszi
+ * @date 2017
+ * Helper class for optimiser settings.
+ */
+
+#pragma once
+
+namespace dev
+{
+namespace solidity
+{
+
+struct OptimiserSettings
+{
+	/// Reset to defaults.
+	void reset()
+	{
+		runOrderLiterals = false;
+		runPeephole = false;
+		runDeduplicate = false;
+		runCSE = false;
+		runConstantOptimiser = false;
+		constantOptimiserTradeoff = 200;
+	}
+
+	/// Enable basic optimisations.
+	void enableBasic()
+	{
+		reset();
+		runPeephole = true;
+		runDeduplicate = true;
+		runCSE = true;
```

Comment:
This applies to all the comments regarding this part: this is only a 1-to-1 copy of what is happening today, wanted to keep that so for this PR. Happy to change these though separately.